File: Backend/Entities/SurveyDetail.cs

using Newtonsoft.Json;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApi.Entities
{
    public class SurveyDetail
    {
        public int Id { get; set; }

        [ForeignKey("Question")]
        public int QuestionId { get; set; }
        
        [ForeignKey("FieldOfAction")]
        public int FieldOfActionId { get; set; }        
        
        [ForeignKey("Survey")]
        public int SurveyId { get; set; }
        public ICollection<SurveyAnswer> SurveyAnswers { get; set; }

    }
}

File: Frontend/src/app/_models/surveyDetail.ts
import { SurveyAnswer } from './surveyAnswer';

export class SurveyDetail {
    id: number;
    questionId: number;
    fieldOfActionId: number;
    surveyId: number;
    surveyAnswers: SurveyAnswer[] = [];

    constructor(questionId: number, fieldOfActionId: number, surveyId: number, surveyAnswers: SurveyAnswer[]) {
        this.questionId = questionId;
        this.fieldOfActionId = fieldOfActionId;
        this.surveyId = surveyId;
        this.surveyAnswers = surveyAnswers;
    }
}
